toucan_connectors/rok/rok_connector.py

```diff
@@ -12,35 +18,117 @@ class RokDataSource(ToucanDataSource):
     filter: str = FilterSchema
 
 
+class InvalidJWTError(Exception):
+    """
+    Raised when the response from ROK is not JSON deserializable,
+    it occurs when our request is invalid because JWT was not validated
+    by ROK
+    """
+
+
+class InvalidUsernameError(Exception):
+    """
+    Raised when the connector tries to be authenticated
+    with an unrecognized username in the JWT claims
+    """
+
+
 class RokConnector(ToucanConnector):
     data_source_model: RokDataSource
-
     host: str
     username: str
-    password: str
+    password: str = None
+    secret: str = None
```

Comment:
I have a doubt on this, shouldn't the secret be retrieved from env variables ?

Reply:
It will!
We will advise to use some variables to do this, like: `secret: {{ secrets.my_rok_secret_for_db_a }}`